common/testing/taskpoller/taskpoller.go

```diff
@@ -228,7 +228,7 @@ func (p *workflowTaskPoller) pollTask(
 	if err != nil {
 		return nil, err
 	}
-	if resp == nil {
+	if resp == nil || resp.TaskToken == nil {
```

Comment:
Good catch! We had that in before and Alex said it might not be needed. I agree, we should add it back!